### PR TITLE
feat: Implement did anchor store

### LINKED ISSUE #226: Implement did anchor reference store
Implement did anchor reference store

### PR DESCRIPTION
**Pull request by sandrask**
Implement did anchor store.

Closes #226

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

**Review by bstasyszyn — APPROVED** [review]

File: pkg/store/didanchor/store.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	for i, a := range anchorBytes {
+		if a == nil {
+			anchors[i] = ""
+		} else {
+			anchors[i] = string(a)
+		}
+	}
+
+	logger.Debugf("retrieved latest anchors%s for suffixes%s", anchors, suffixes)
```
Comment (DRK3) [inline pkg/store/didanchor/store.go:77]: Missing spaces between `%s`

**Review by fqutishat — APPROVED** [review]

File: pkg/store/didanchor/store_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		s, err := New(provider)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "failed to open did anchor store: open store error")
+		require.Nil(t, s)
+	})
+}
+
+func TestStore_Put(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		provider := &mocks.Provider{}
```
Comment (DRK3) [inline pkg/store/didanchor/store_test.go:42]: For success cases, I suggest using the real in-memory store implementation (found [here](https://github.com/hyperledger/aries-framework-go/blob/main/component/storageutil/mem/mem.go)). If storage has to be updated later, I've found it easier when dealing with real storage implementations. It also makes the unit tests act closer to the real thing.

File: pkg/store/didanchor/store_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		require.NoError(t, err)
+
+		anchors, err := s.Get([]string{"suffix-1", "suffix-2"})
+		require.NoError(t, err)
+		require.Equal(t, "cid-1", anchors[0])
+		require.Equal(t, "cid-2", anchors[1])
+	})
+
+	t.Run("success - with cid not found case", func(t *testing.T) {
+		store := &mocks.Store{}
```
Comment (DRK3) [inline pkg/store/didanchor/store_test.go:86]: You can also set up this error scenario using the real in-memory implementation without mocking

**Review by DRK3 — APPROVED** [review]
Minor, optional comments

File: pkg/store/didanchor/store.go (hunk 1 above)
Comment (sandrask) [reply pkg/store/didanchor/store.go:77]: This is on purpose, looks good in log.

File: pkg/store/didanchor/store_test.go (hunk 2 above)
Comment (sandrask) [reply pkg/store/didanchor/store_test.go:42]: Sure, I will use in-memory store for success scenario.

File: pkg/store/didanchor/store_test.go (hunk 3 above)
Comment (sandrask) [reply pkg/store/didanchor/store_test.go:86]: Same as above, this is success scenario.

File: pkg/store/didanchor/store_test.go (hunk 2 above)
Comment (sandrask) [reply pkg/store/didanchor/store_test.go]: Fixed

**Pull request merged**